### PR TITLE
airbyte-protocol: remove `supportsNormalization` and `supportsDbt` from `ConnectorSpecification`

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/2536
We are removing the `supportsNormalization` and `supportsDbt` from the connector specification and moved these metadata to destination definitions. (https://github.com/airbytehq/airbyte/pull/21005)

## How
* Remove these two fields from the protocols `ConnectorSpecification` model in `airbyte_protocol.yaml`
* Update mocks using `withSupportsDbt` or `withSupportsNormalization`
* Update the CDK models

### CODE REVIEW

**Comment by alafanechere** [comment]
@gosusnp `testDowngrade` does not pass. I guess this might have something to do with protocol versioning / migration. Could you please tell me what I could do to make this test pass or point me to documentation about protocol updates with migrations?

**Comment by cgardens** [comment]
woohoo!

**Comment by gosusnp** [comment]
@alafanechere, looking at the test failures, I'd expect it comes from the input of the test. The fields that were removed are now in the `additionalProperties` field. Most likely because the updated model doesn't know about them.

In terms of protocol version, this PR removes fields so it would be a breaking change and should result in a major bump. Generally, I would recommend to delay this change to avoid bumping majors with only clean up of old fields.
One suggestion is that since we are currently working on getting v1 of the protocol with new data types out. We could take that opportunity to remove those fields as well.

**Comment by alafanechere** [comment]
Thank you @gosusnp for the details
> Generally, I would recommend to delay this change to avoid bumping majors with only clean up of old fields.
I'm not entirely sure about what you're suggesting here. 
Should the other changes I made on this PR (the mock updates) happen before we make the protocol change?

> One suggestion is that since we are currently working on getting v1 of the protocol with new data types out. We could take that opportunity to remove those fields as well.

Is this happening on another PR on which I could push these changes?

File: airbyte-protocol/protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml [hunk 1]
```diff
@@ -452,16 +452,6 @@ definitions:
...
-      # Destination Connectors Properties
-      # Normalization is currently implemented using dbt, so it requires `supportsDBT` to be true for this to be true.
-      supportsNormalization:
-        description: If the connector supports normalization or not.
-        type: boolean
-        default: false
-      supportsDBT:
-        description: If the connector supports DBT or not.
-        type: boolean
-        default: false
```
Comment (evantahler) [inline airbyte-protocol/protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml:464]: There are probably some related updates to make in https://github.com/airbytehq/airbyte/blob/master/docs/understanding-airbyte/airbyte-protocol.md#actor-specification

Comment (alafanechere) [reply airbyte-protocol/protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml:464]: Done, thanks for the suggestion.

**Comment by evantahler** [comment]
@gosusnp can you help me understand why this is a breaking change?  Perhaps this is about message validation.  As supportsDbt and supportsNormalization still required platform changes to work, we don't expect OSS users to have made connectors that made use of these flags.  Ideally we can remove them without breaking things for users.

* Do we need to change `additionalProperties:true` in all the destination specs to avoid parse errors?  I don't think so, because the specs themselves are definitions in jsonSchema, maybe?
* Do we need to change the model for ConnectorSpecification?  [This](https://github.com/airbytehq/airbyte/blob/master/airbyte-config/config-models/src/main/resources/types/StandardGetSpecOutput.yaml) might imply that perhaps we should change `additionalProperties:true` here, so that when a connector returns the now-removed supportsDbt property, we don't fail validation

**Comment by gosusnp** [comment]
@evantahler, my comment is purely on the nature of the change, removing a field should fall under a breaking change. Now, that doesn't mean it will actually break something.

I think the default for `additionalProperties` is `true` so it may be redundant to add it everywhere. The test Augustin made shows that we parsed those attributes into the `additionalProperties` so we are not going to reject the spec. Given the breath of connectors, we should benefit from being more fault resilient when it comes to parsing the messages and probably should only fail on attributes that are key to a correct replication.

`StandardGetSpecOutput` is a different use case, I do not think those objects should be considered public. Being stricter here enables us to catch our internal errors sooner. Some of those objects currently bleed into the connectors mostly from how we test, after recent discussions, I am leaning towards providing dedicated tooling for this purpose which would remove this dependency.

This change might be safe, but a goal of versioning was to take the guess work out of it. Unless we have a strong argument for having to clean it now, I would generally recommend to flag for removal and remove in the next version.

**Review by sherifnada — APPROVED** [review]

**Comment by alafanechere** [comment]
> This change might be safe, but a goal of versioning was to take the guess work out of it. Unless we have a strong argument for having to clean it now.


My only argument to clean it now is, but I'm not sure it's a strong one :smile :
- Java connectors rely on the protocols models to build the `spec` output. If we're not removing these fields in the protocol they will fall back to the default `false` value, which will be misleading (https://github.com/airbytehq/airbyte/pull/21306). We'll have `spec` outputs with `supportsNormalization: false` on destination bigquery. It won't break the connector as the normalization support is now stored in the definition, but it will create misleading `spec` outputs for the connector.

>  I would generally recommend to flag for removal and remove in the next version

How can I flag these change to be made in the next protocol version?

**Comment by alafanechere** [comment]
As @gosusnp mentions it's a breaking change I'm closing this PR and adding the original issue to this [EPIC](https://github.com/airbytehq/airbyte/issues/14276) to flag it for removal in the next major version.

**Pull request closed without merging**